ksql-execution/src/main/java/io/confluent/ksql/execution/plan/StreamSource.java

```diff
@@ -63,22 +99,359 @@ public boolean equals(final Object o) {
     if (o == null || getClass() != o.getClass()) {
       return false;
     }
-    final StreamSource<?> that = (StreamSource<?>) o;
+    final StreamSource that = (StreamSource) o;
     return Objects.equals(properties, that.properties)
         && Objects.equals(topicName, that.topicName)
         && Objects.equals(formats, that.formats)
-        && Objects.equals(timestampPolicy, that.timestampPolicy)
-        && offsetReset == that.offsetReset;
+        && Objects.equals(timestampPolicy, that.timestampPolicy);
   }
 
   @Override
   public int hashCode() {
-
-    return Objects.hash(properties, topicName, formats, timestampPolicy, offsetReset);
+    return Objects.hash(
+        properties,
+        topicName,
+        formats,
+        timestampPolicy
+    );
   }
 
   @Override
-  public KStream<K, GenericRow> build(final StreamsBuilder builder) {
-    throw new UnsupportedOperationException();
+  public KStream<K, GenericRow> build(final KsqlQueryBuilder queryBuilder) {
+    // this hack will go away soon. it's just glue until we don't need keySerde in
+    // SchemaKStream/SchemaKTable anymore
+    return buildWithSerde(queryBuilder).getRight();
+  }
+
+  private Consumed<K, GenericRow> buildSourceConsumed(
+      final Serde<K> keySerde,
+      final Serde<GenericRow> valueSerde) {
+    final TimestampExtractor timestampExtractor = timestampPolicy.create(timestampIndex);
+    final Consumed<K, GenericRow> consumed = functions.consumedFactory
+        .apply(keySerde, valueSerde)
+        .withTimestampExtractor(timestampExtractor);
+    if (offsetReset.isPresent()) {
+      return consumed.withOffsetResetPolicy(offsetReset.get());
+    }
+    return consumed;
+  }
+
+  private KStream<K, GenericRow> buildKStream(
+      final KsqlQueryBuilder queryBuilder,
+      final Consumed<K, GenericRow> consumed,
+      final ValueMapperWithKey<K, GenericRow, GenericRow> mapper) {
+    return queryBuilder.getStreamsBuilder()
+        // 1. Create a KStream on the changelog topic.
+        .stream(topicName, consumed)
+        // 2. mapValues to add the ROWKEY column
+        .mapValues(mapper)
+        // 3. transformValues to add the ROWTIME column. transformValues is required to access the
+        //    streams ProcessorContext which has the timestamp for the record.
+        .transformValues(functions.timestampTransformer);
+  }
+
+  public Pair<KeySerde<K>, KStream<K, GenericRow>> buildWithSerde(
```

Comment:
This is a temporary glue function. Once `structured` is switched over to build execution plans we won't need to pass around the actual serdes in the engine layer, and this can go away.